### PR TITLE
[chore] Schema updates

### PR DESCRIPTION
**Pull request by kgrhartlage**
### Description

- update several descriptions for clarity
- require `type` setting for all fields
- prevent setting both `unit` and `kind` together for fields of type quantity
- update `.gitignore` to include files that match the schema association pattern defined the in the VS Code settings
- update README with links

These changes to the schema definitions have already been approved on the server repo.

### CODE REVIEW

File: dist/workflow_step_template_schema.json [hunk 1]
```diff
@@ -1329,10 +1331,38 @@
...
+                                },
+                                {
+                                    "required": [
+                                        "kind"
+                                    ]
+                                }
+                            ]
+                        }
+                    }
+                ],
```
Comment (chudoklates) [inline dist/workflow_step_template_schema.json:1365]: This is the absolute strangest XOR statement that I've seen

File: src/schemata/definitions/workflow_template/changeReason.yml [hunk 2]
```diff
@@ -1,17 +1,14 @@
 ---
 title: Change Reason
 description: |
-  A flag indicating that overwriting an existing field value requires a reason and prompts the user with a reason dialog. The reason is recorded along with the new field value. If no reason is submitted, the field is not updated.
-
-  Updating a field that has no value with a value does not require a reason. A field update that changes the value to `null`, however, does require a reason. Consider the following example:
```
Comment (chudoklates) [inline src/schemata/definitions/workflow_template/changeReason.yml:6]: IIUC, this is no longer the truth? I.E. change reasons must be provided every time there is a change after the initial value has been set?

Comment (kgrhartlage) [reply src/schemata/definitions/workflow_template/changeReason.yml:6]: > The first update to a field's value never requires a reason, any further updates always do.

Doesn't matter if the field has a `defaultValue` or not to begin with, it's about the first and all subsequent updates

Comment (kgrhartlage) [reply src/schemata/definitions/workflow_template/changeReason.yml:6]: While the text description was correct, I just realized that the step-by-step example was incorrect. I pushed the change.

**Review by chudoklates — APPROVED** [review]
LGTM 👍

**Pull request merged**